src/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -10,14 +10,20 @@ open System.Xml.Linq
 open Microsoft.Build.Framework
 open Microsoft.Build.Utilities
 
-type FSharpEmbedResXSource() =
-    let mutable _buildEngine: IBuildEngine MaybeNull = null
-    let mutable _hostObject: ITaskHost MaybeNull = null
+type FSharpEmbedResXSource() as this =
+    inherit Task()
     let mutable _embeddedText: ITaskItem[] = [||]
     let mutable _generatedSource: ITaskItem[] = [||]
     let mutable _outputPath: string = ""
     let mutable _targetFramework: string = ""
 
+    let failTask fmt =
```

Comment:
Probably another personal preference but I never saw this `ksprintf` thing. The name is not intuitive so I believe readers would have to look it up. Thus I'd rather stay with well-known functions.

Reply:
This is the way to create a partially applied printfn-like function (but with different side effect in the end, in this case LogError). Without it, it is not possible to create a function that would nicely curry the formatting string and any number of possible arguments.

See e.g. https://stackoverflow.com/questions/5569909/how-do-i-create-an-f-function-with-a-printf-style-logging-argument

Reply:
Yeah well anyway, not a big deal. Printing is printing, some flavors are details :)

Reply:
This function is equivalent to

```fsharp
let failTask fmt =
    this.Log.LogError (sprintf fmt)
    raise TaskFailed
```

but because the `printf` family of functions is magic, we can't do it so `ksprintf` comes to the rescue.